packages/shared/components/Overlay.tsx

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import ReactDOM from "react-dom";
+
+interface OverlayProps {
+  overlayRoot?: HTMLElement;
+  className?: string;
+}
+
+class Overlay extends React.Component<OverlayProps> {
+  public static defaultProps: Partial<OverlayProps> = {
+    overlayRoot: document.body
+  };
+
+  private el: HTMLElement;
+
+  constructor(props) {
+    super(props);
+    this.el = document.createElement("div");
+  }
+
+  componentDidMount() {
+    if (this.props.overlayRoot) {
+      this.props.overlayRoot.appendChild(this.el);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.props.overlayRoot) {
+      this.props.overlayRoot.removeChild(this.el);
+    }
+  }
+
+  getContents() {
+    const { className, children } = this.props;
+
+    return <div className={className}>{children}</div>;
+  }
+
+  render() {
+    return ReactDOM.createPortal(this.getContents(), this.el);
```

Comment:
that wouldn't work with react@15 we're using in the dcos-ui, right?

Reply:
It will not. The decision was made between mike and rodney to wait until we upgrade react to use these components in dcos-ui. Interestingly, ui-kit package.json already stipulates react 16 so how does any of this work?? 🙃